### PR TITLE
EVA-2451 - Fix Variant-remapping process from testing

### PR DESCRIPTION
**Pull request by tcezard**
Couple of fixes from testing the pipeline. Possible that the mongo read preference should be reverted once MongoDB is upgraded
Added parsing of the logs and yaml to record counts

### CODE REVIEW

File: variant-remapping-automation/remapping_automation.py [hunk 1]
```diff
@@ -69,7 +73,7 @@ def write_remapping_process_props_template(template_file_path):
 spring.data.mongodb.password={mongo_pass}
 
 spring.data.mongodb.authentication-database=admin
-mongodb.read-preference=secondaryPreferred
+mongodb.read-preference=primaryPreferred
```
Comment (sundarvenkata-EBI) [inline variant-remapping-automation/remapping_automation.py:76]: A TODO could be added here to indicate that this needs to be reverted post 4.0 migration.

File: variant-remapping-automation/remapping_automation.py [hunk 2]
```diff
@@ -83,84 +87,114 @@ def get_metadata_conn():
...
+            if nb_variant_extracted is not None:
+                set_statements.append(f"nb_variant_extracted = {nb_variant_extracted}")
+            if nb_variant_remapped is not None:
+                set_statements.append(f"nb_variant_remapped = {nb_variant_remapped}")
+            if nb_variant_ingested is not None:
+                set_statements.append(f"nb_variant_ingested = {nb_variant_ingested}")
+
+        if set_statements:
+            query = ('UPDATE remapping_progress '
+                     'SET '+' '.join(set_statements) + ' '
```
Comment (sundarvenkata-EBI) [inline variant-remapping-automation/remapping_automation.py:166]:
```suggestion
                     'SET '+','.join(set_statements) + ' '
```

**Review by sundarvenkata-EBI — APPROVED** [review]
Looks good! Will there be tests added later for the newly added functionality?

File: variant-remapping-automation/tests/test_remapping_automation.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import os
+import unittest
+
+from remapping_automation import count_variants_extracted
+
+
+class TestRemmapingAutomation(unittest.TestCase):
```
Comment (apriltuesday) [inline variant-remapping-automation/tests/test_remapping_automation.py:7]:
```suggestion
class TestRemappingAutomation(unittest.TestCase):
```

File: variant-remapping-automation/tests/test_remapping_automation.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+    def test_count_variants_extracted(self):
+        log_file = os.path.abspath(os.path.join(os.path.dirname(__file__), 'ressources', 'vcf_extractor.log'))
```
Comment (apriltuesday) [inline variant-remapping-automation/tests/test_remapping_automation.py:10]: does this file need to be added to git?

**Review by apriltuesday — APPROVED** [review]

File: variant-remapping-automation/remapping_automation.py (hunk 2 above)
Comment (tcezard) [reply variant-remapping-automation/remapping_automation.py]: Done

File: variant-remapping-automation/tests/test_remapping_automation.py (hunk 4 above)
Comment (tcezard) [reply variant-remapping-automation/tests/test_remapping_automation.py:10]: Done

File: variant-remapping-automation/remapping_automation.py (hunk 1 above)
Comment (tcezard) [reply variant-remapping-automation/remapping_automation.py:75]: Done

**Pull request merged**